torch/distributed/tensor/parallel/style.py

```diff
@@ -334,8 +509,30 @@ class RowwiseParallel(ParallelStyle):
     We assume the input to be a sharded :class:`DTensor` and output to be a :class:`torch.Tensor`.
     """
 
-    def __init__(self, _prepare_input=make_input_shard_1d_last_dim, _prepare_output=make_output_tensor) -> None:
-        super().__init__(_prepare_input, _prepare_output)
+    def __init__(
+        self,
+        input_layouts=Shard(-1),
+        output_layouts=Replicate(),
+        use_local=True,
+        _prepare_input=None,
+        _prepare_output=None,
+    ) -> None:
+        super().__init__(
+            input_layouts=input_layouts,
+            output_layouts=output_layouts,
+            use_local=use_local,
+            _prepare_input=_prepare_input
+            if _prepare_input is not None
+            else _get_prepare_input(
+                input_layouts,
+                [Shard(-1)] * len(input_layouts)  # type: ignore[arg-type]
```

Comment:
I don't quite get this, if `isinstance(input_layouts, tuple)`, we should just use that directly? why need to duplicate the input layouts when generate the prepare_input_fn?

Reply:
No, because input_layouts can be not always [Shard(-1), Shard(-1)]?